redhat-appstudio/service-provider-integration-operator
# make binding display an error in case of invalid service provider url

Signed-off-by: Pavol Baran <pbaran@redhat.com>

### What does this PR do?
In case of invalid service provider URL (derived from repoUrl) this PR makes binding to display error massage and prevents creation of spiaccesstoken for this binding.

### Screenshot/screencast of this PR

![image](https://user-images.githubusercontent.com/73115616/190964174-cfc83873-fb57-4f1f-a9ed-5aadc41ca999.png)
![image](https://user-images.githubusercontent.com/73115616/190964327-4eef0dc2-e83f-4cc7-9372-e20b1876b076.png)




### What issues does this PR fix or reference?

fixes: https://issues.redhat.com/browse/SVPI-183

### How to test this PR?

1. Deploy SPI with operator image `quay.io/pbaran/spio:SVPI-183`
2. Create binding with repoUrl that is not valid such as: "https://abc."
3. Check that binding is in error phase, displays an error message and that no new spiaccesstoken is created nor linked with this binding.
4. Create binding with valid repoUrl e.g. `kubectl apply -f samples/binding-read-repo.yaml` and check that no error occured.
